src/wp-includes/class-wp-theme-json.php

```diff
@@ -2298,19 +2325,25 @@ public function merge( $incoming ) {
 				$override_preset = ! static::get_metadata_boolean( $this->theme_json['settings'], $preset['prevent_override'], true );
 
 				foreach ( static::VALID_ORIGINS as $origin ) {
-					$base_path = array_merge( $node['path'], $preset['path'] );
-					$path      = array_merge( $base_path, array( $origin ) );
-					$content   = _wp_array_get( $incoming_data, $path, null );
+					$base_path = $node['path'];
+					foreach ( $preset['path'] as $leaf ) {
+						$base_path[] = $leaf;
+					}
+
+					$path   = $base_path;
```

Comment:
Same question here about why not use `$path = array()`;

Reply:
Based on what the original code does, `$path` needs to include the items in `$base_path`, plus the single `$origin` item, so if we set `$path` to an empty array here, the behavior would change.